src/listeners/listener.interface.ts

```diff
@@ -0,0 +1,8 @@
+import { LavalinkManagerEvents, NodeManagerEvents } from 'lavalink-client';
+
+type FormatEventInterface<T> = {
+	[K in keyof T]: T[K] extends (...args: infer P) => void ? P : never;
+};
+
+export type NecordLavalinkEvents = FormatEventInterface<LavalinkManagerEvents> &
+	FormatEventInterface<NodeManagerEvents>;
```

Comment:
Can you provide some explanation?

Reply:
For the `LavalinkContextOf` (like [ContextOf](https://github.com/necordjs/necord/blob/master/src/context/necord-context.interface.ts#L41)) to work correctly pass event parameters typing , the event interface must have the following format
```TypeScript
export interface Events {
    eventName: [parameterOne: string, parameterTwo: number];
}
```

but the Event [interfaces](https://github.com/Tomato6966/lavalink-client/blob/main/dist/types/structures/Types/Manager.d.ts#L11-L117) of [lavalink-client](https://github.com/Tomato6966/lavalink-client) have this format
```TypeScript
export interface Events {
    'eventName': (parameterOne: string, parameterTwo: number) => void;
}
```

this helper type converts the interface format